test-framework/jacoco/runtime/src/main/java/io/quarkus/jacoco/runtime/DataFileWatch.java

```diff
@@ -0,0 +1,81 @@
+package io.quarkus.jacoco.runtime;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.function.BooleanSupplier;
+import java.util.function.LongSupplier;
+
+class DataFileWatch {
+    static final int ABORT_TIMEOUT_MILLIS = 10000;
+    private final LongSupplier clock;
+    private final Path datafile;
+    private final long initialDataFileSize;
+    private final BooleanSupplier jacocoFinished;
+
+    DataFileWatch(Path datafile, BooleanSupplier jacocoFinished) {
+        this(System::currentTimeMillis, datafile, jacocoFinished);
+    }
+
+    // Constructor for tests
+    DataFileWatch(LongSupplier clock, Path datafile, BooleanSupplier jacocoFinished) {
+        this.clock = clock;
+        this.datafile = datafile;
+        this.jacocoFinished = jacocoFinished;
+
+        // Remember the initial size of the data file. If the data file does not already exist,
+        // the value will be -1, otherwise (if `JacocoConfig.reuseDataFile` is true and the file
+        // exists) the "old" size of the data file.
+        // In other words: Jacoco wrote the data file, if the file size is different from the
+        // value of `initialDataFileSize`.
+        try {
+            this.initialDataFileSize = dataFileSize();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    boolean waitForDataFile() throws IOException, InterruptedException {
+        // The jacoco data is also generated by a shutdown hook.
+        // Wait at most 10 seconds until the report data file appears, also wait as long as the report data
+        // file grows, giving jacoco some "grace time" to write the next chunk of data, exit the loop when
+        // either the "abort timeout" is hit or the jacoco thread is no longer running.
+        long abortTime = nextAbortTime();
+        long fileSize;
+        for (long previousFileSize = initialDataFileSize;; previousFileSize = fileSize) {
```

Comment:
nit: add a space between `;;` for readability?

Reply:
... or perhaps convert to `while(true)`?

Reply:
Nope, code's automagically formatted